dbusmock/templates/systemd-base.py

```diff
@@ -0,0 +1,111 @@
+'''systemd mock template base
+'''
+
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the Free
+# Software Foundation; either version 3 of the License, or (at your option) any
+# later version.  See http://www.gnu.org/copyleft/lgpl.html for the full text
+# of the license.
+
+__author__ = 'Jonas Ådahl'
+__copyright__ = '(c) 2021 Red Hat'
+
+from gi.repository import GLib
+import dbus
+
+from dbusmock import MOCK_IFACE, mockobject
+
+BUS_PREFIX = 'org.freedesktop.systemd1'
+PATH_PREFIX = '/org/freedesktop/systemd1'
+
+
+BUS_NAME = BUS_PREFIX
+MAIN_OBJ = PATH_PREFIX
+MAIN_IFACE = BUS_PREFIX + '.Manager'
+UNIT_IFACE = BUS_PREFIX + '.Unit'
+
+
+def load_base(mock, _parameters):
+    mock.next_job_id = 1
+    mock.units = {}
+
+    mock.AddServiceMethod(MAIN_IFACE, 'StartUnit', StartUnit)
+    mock.AddServiceMethod(MAIN_IFACE, 'StartTransientUnit', StartTransientUnit)
+    mock.AddServiceMethod(MAIN_IFACE, 'StopUnit', StopUnit)
+    mock.AddServiceMethod(MAIN_IFACE, 'GetUnit', GetUnit)
+    mock.AddProperties(MAIN_IFACE, {'Version': 'v246'})
+
+    mock.AddServiceMethod(MOCK_IFACE, 'AddMockUnit', AddMockUnit)
+
+
+def escape_unit_name(name):
+    for s in ['.', '-']:
+        name = name.replace(s, '_')
+    return name
+
+
+def emit_job_new_remove(self, job_id, job_path, name):
```

Comment:
Given this is not an object "class" function, maybe better not to use `self` here but just `mock` maybe?

Reply:
Hrm, none of the functions are in a class and use `self`..

Reply:
Not technically, but practically we consider them as part of the "mock dbus class", but not a big deal, I just was finding it clearer.

Reply:
It's now `mock`.